settings/rest-api.php

```diff
@@ -131,6 +132,33 @@ function register_rest_routes() : void {
 			),
 		),
 	);
+
+	register_rest_route(
+		'wporg-two-factor/1.0',
+		'/generate-svn-password',
+		array(
+			'methods'  => WP_REST_Server::EDITABLE,
+			'callback' => function( $request ) {
+				$user = get_userdata( $request['user_id'] );
+				return [
+					'svn_password' => set_svn_password( $user->ID )
+				];
+			},
+			'permission_callback' => function( $request ) {
+				return Two_Factor_Core::rest_api_can_edit_user_and_update_two_factor_options( $request['user_id'] );
+			},
+			'args' => array(
+				'user_id' => array(
+					'required' => true,
+					'type'     => 'number',
+					'sanitize_callback' => 'absint',
+					'validate_callback' => function( $user_id ) {
+						return get_userdata( $user_id ) instanceof WP_User;
+					},
+				),
```

Comment:
Do we allow user impersonation in WP.org? If not, maybe we can just use the current user id?

Reply:
We do, although not how you may have invisioned it here.

Currently a super-admin can modify another users account data, that's why `user_id` is passed to the various 2FA endpoints.

I'm not a super-fan of it, but it's how the other endpoints in the plugin operate, so it made sense to duplicate that here.

An example of where this IS needed, is a case of accounts that cannot be logged into, system-level automation accounts that we need a SVN password for.  However, we do have the CLI commands that can be used for that.

It also allows for super-admins to be able to quickly re-generate a new svn password for a user to lock them out (although, that's not something we'd likely make use of).

Happy to change this to throw a error along the lines of `You can only re-generate a password for yourself` if you feel strongly.